### PR TITLE
Feature/milestone feature

### PR DESCRIPTION
**Pull request by VladislavKostyukevich**
# PR Details

Milestone List
Milestone view

## Related Issue

  - Milestone View: Add functionality that displays the latest results of the tests from the test runs -> [View Issue](https://github.com/aquality-automation/aquality-tracking/issues/11)

## How Has This Been Tested

See automated tests

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: e2e/elements/charts/resultPie.element.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import allResults from '../../data/results.json';
+
+const results: PieChartSections = allResults;
+
+export class ResultPieChart extends PieChartBase {
+    constructor(locator: Locator | ElementFinder) {
+        super(locator);
+    }
+
+    public clickNone() {
```
Comment (VladislavKostyukevich) [inline e2e/elements/charts/resultPie.element.ts]: This will not work!

File: e2e/elements/charts/resolution.Pie.element.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import allResolutions from '../../data/resolutions.json';
+
+const resolutions: PieChartSections = allResolutions.global;
+
+export class ResolutionPieChart extends PieChartBase {
+    constructor(locator: Locator | ElementFinder) {
+        super(locator);
+    }
+
+    public clickNone() {
```
Comment (VladislavKostyukevich) [inline e2e/elements/charts/resolution.Pie.element.ts]: This will not work!

File: e2e/helpers/project.helper.ts [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+            logger.error(err.message);
+        }
+    }
+
+    public async openProject() {
+        await apiTokenAdministration.menuBar.clickLogo();
+        return projectList.openProject(this.project.name);
+    }
+
+    public async  dispose() {
```
Comment (VladislavKostyukevich) [inline e2e/helpers/project.helper.ts]: extra space

File: e2e/helpers/project.helper.ts [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        await logIn.logInAs(this.admin.user_name, this.admin.password);
+        await projectList.isOpened();
+        await projectList.removeProject(this.project.name);
+    }
+
```
Comment (VladislavKostyukevich) [inline e2e/helpers/project.helper.ts]: blank line

File: e2e/pages/testrun/view.po/index.ts [hunk 5]
```diff
@@ -64,12 +62,12 @@ export class TestRunView extends BasePage {
     return elements.resultsTable.rightClickCell(columns.failReason, failReason, columns.failReason);
   }
 
-  async clickResultPieChartSection(id: number) {
-    return browser.executeScript(pieChartClickSectionScript, elements.resultsChart, id);
+  async clickResultPieChartSection() {
```
Comment (VladislavKostyukevich) [inline e2e/pages/testrun/view.po/index.ts]: Rename method

File: e2e/pages/testrun/view.po/index.ts [hunk 6]
```diff
@@ -64,12 +62,12 @@ export class TestRunView extends BasePage {
...
+    return elements.resultsChart.clickPassed();
   }
 
-  async clickResolutionPieChartSection(id: number) {
-    return browser.executeScript(pieChartClickSectionScript, elements.resolutionsChart, id);
+  async clickResolutionPieChartSection() {
```
Comment (VladislavKostyukevich) [inline e2e/pages/testrun/view.po/index.ts]: Rename method

File: e2e/specs/administration/projectSettings/projectSettings.searchPattern.spec.ts [hunk 7]
```diff
@@ -31,48 +30,31 @@ const builds = {
...
-    const projectsList: ProjectList = new ProjectList();
-    const userAdministration: UserAdministration = new UserAdministration();
-    const permissionsAdministration: PermissionsAdministration = new PermissionsAdministration();
-    const projectSettingsAdministration: ProjectSettingsAdministration = new ProjectSettingsAdministration();
-    const projectView: ProjectView = new ProjectView();
-    const testRunList: TestRunList = new TestRunList();
-    const testRunView: TestRunView = new TestRunView();
-
+    const projectHelper: ProjectHelper = new ProjectHelper();
     const project: Project = projects.customerOnly;
```
Comment (VladislavKostyukevich) [inline e2e/specs/administration/projectSettings/projectSettings.searchPattern.spec.ts]: Remove this

File: e2e/specs/administration/projectSettings/projectSettings.spec.ts [hunk 8]
```diff
@@ -23,46 +21,33 @@ const notEditorExamples = {
...
 describe('Administartion:', () => {
-    const logInPage: LogIn = new LogIn();
-    const projectsList: ProjectList = new ProjectList();
-    const userAdministration: UserAdministration = new UserAdministration();
-    const permissionsAdministration: PermissionsAdministration = new PermissionsAdministration();
-    const projectSettingsAdministration: ProjectSettingsAdministration = new ProjectSettingsAdministration();
-    const notFound: NotFound = new NotFound();
-
+    const projectHelper: ProjectHelper = new ProjectHelper();
     const project: Project = projects.customerOnly;
```
Comment (VladislavKostyukevich) [inline e2e/specs/administration/projectSettings/projectSettings.spec.ts]: Remove this

File: e2e/specs/api/publicApi.spec.ts [hunk 9]
```diff
@@ -1,57 +1,44 @@
-import { LogIn } from '../../pages/login.po';
```
Comment (VladislavKostyukevich) [inline e2e/specs/api/publicApi.spec.ts]: Use editor API from ProjectHelper

File: e2e/elements/charts/pieBase.element.ts [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+    constructor(locator: Locator | ElementFinder) {
+        super(locator);
+    }
+
+    protected async clickChartSection(section: PieChartSection) {
+        await browser.executeScript(this.clickSectionScript, this.element, section.index);
+        logger.info(`${section.name} section of Pie Chart was clicked.`);
+        return section.name;
+    }
+
```
Comment (knysh) [inline e2e/elements/charts/pieBase.element.ts]: remove redundant space

File: e2e/elements/smartTable.element/index.ts [hunk 11]
```diff
@@ -23,6 +26,27 @@ export class SmartTable extends BaseElement {
...
+            message: ''
+        };
+        const actualTableCSV = await this.getCSV();
+        const expectedTableCSV = await testData.readAsString(pathToCSV);
+        const comparisonResult = compareCSVStrings(actualTableCSV, expectedTableCSV, true);
+        if (comparisonResult.missedFromActual.length > 0) {
+            result.result = false;
+            result.message = `Not all actual results are in expected list:\n${comparisonResult.missedFromActual.join('\n')}`;
+        }
+        if (comparisonResult.missedFromActual.length > 0) {
```
Comment (knysh) [inline e2e/elements/smartTable.element/index.ts]: as I understand it should be 'missedFromExpected' here

File: e2e/specs/api/publicApi.spec.ts (hunk 9 above)
Comment (knysh) [reply e2e/specs/api/publicApi.spec.ts]: fixed

**Pull request merged**